# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/test/mochitest/test_notification_tag.html
# Commit: ed71fdc2d78d
# Full Hash: ed71fdc2d78dec085a3d153220bba4aeda89992b
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2025-03-05 04:28:59
# Regressor Bug: 1951379
# File Overlap Count: 1
# Description:
#   Bug 1951379 - Part 3: Add NotificationStorageEntry r=asuth
#   
#   This skips handling `vibrate` for now, as we don't ship it and it also complicates test setup.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D240213
# ==============================================================================

diff -r 61ab7888abc4 -r ed71fdc2d78d dom/notification/test/mochitest/test_notification_tag.html
--- a/dom/notification/test/mochitest/test_notification_tag.html	Tue Mar 04 18:00:17 2025 +0000
+++ b/dom/notification/test/mochitest/test_notification_tag.html	Tue Mar 04 18:00:18 2025 +0000
@@ -39,15 +39,9 @@
 
     const storageGet = (origin) => {
       const { promise, resolve } = Promise.withResolvers();
-      const ids = [];
-      notificationStorage.get(origin, "", "", {
-        handle(id) {
-          ids.push(id);
-        },
-        done() {
-          resolve(ids);
-        }
-      })
+      notificationStorage.get(origin, "", "", notifications => {
+        resolve(notifications.map(({ id }) => id));
+      });
       return promise;
     };
 